CMakeLists.txt

```diff
@@ -73,6 +73,22 @@ option(CREATE_DOC "Whether or not to create doxygen doc target." OFF)
 list(APPEND CMAKE_PREFIX_PATH $ENV{ROOTSYS})
 set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
 find_package(ROOT REQUIRED COMPONENTS RIO Tree)
+
+# Check that root is compiled with a modern enough c++ standard
+get_target_property(ROOT_COMPILE_FEATURES ROOT::Core INTERFACE_COMPILE_FEATURES)
+if (NOT "cxx_std_17" IN_LIST ROOT_COMPILE_FEATURES)
```

Comment:
While we need cxxstd17 for podio, I believe root may be compiled with cxxstd14 for this to work also. And root builds with cxxstd14 are still very common, see for example the discussion here: https://github.com/root-project/root-docker/issues/35 But I'll double-check!

Reply:
I am in favour of insisting on a consistent standard.